openmetadata-ui/src/main/resources/ui/src/components/common/TestConnection/TestConnection.tsx

```diff
@@ -223,57 +232,71 @@ const TestConnection: FC<TestConnectionProps> = ({
         return;
       }
 
-      /**
-       * fetch workflow repeatedly with 2s interval
-       * until status is either Failed or Successful
-       */
-      intervalId = toNumber(
-        setInterval(async () => {
-          setProgress((prev) => prev + TEST_CONNECTION_PROGRESS_PERCENTAGE.ONE);
-          const workflowResponse = await getWorkflowData(response.id);
-          const { response: testConnectionResponse } = workflowResponse;
-          const { status: testConnectionStatus, steps = [] } =
-            testConnectionResponse || {};
-
-          const isWorkflowCompleted = WORKFLOW_COMPLETE_STATUS.includes(
-            workflowResponse.status as WorkflowStatus
-          );
-
-          const isTestConnectionSuccess =
-            testConnectionStatus === StatusType.Successful;
-
-          if (!isWorkflowCompleted) {
-            return;
-          }
-
-          setProgress(TEST_CONNECTION_PROGRESS_PERCENTAGE.HUNDRED);
-          if (isTestConnectionSuccess) {
-            setTestStatus(StatusType.Successful);
-            setMessage(TEST_CONNECTION_SUCCESS_MESSAGE);
-          } else {
-            const isMandatoryStepsFailing = steps.some(
-              (step) => step.mandatory && !step.passed
-            );
-            setTestStatus(
-              isMandatoryStepsFailing ? StatusType.Failed : 'Warning'
+      // return a promise that wraps the interval and handles errors inside it
+      const workflowPromise = new Promise<void>((resolve, reject) => {
```

Comment:
@Ashish8689 , why are we creating a promise here?

Reply:
since, we are calling the API after every some interval, so to manage the repeated asynchronous operation (i.e., polling the status), have used promise here to ensure proper error handling as well, if any polling API fails or API aborts, so that can be handled properly with the promise.